exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
 	ManagedControlPlaneFinalizer = "gcpmanagedcontrolplane.infrastructure.cluster.x-k8s.io"
 )
 
+// PrivateCluster defines a private Cluster.
+type PrivateCluster struct {
+	// Enabled defines weather a cluster is private or not. If disabled cluster network is public.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// ControlPlaneCidrBlock is the IP range in CIDR notation to use for the hosted master network. This range must not
+	// overlap with any other ranges in use within the cluster's network. Honored when enabled is true.
+	// +optional
+	ControlPlaneCidrBlock string `json:"controlPlaneCidrBlock,omitempty"`
+
+	// ControlPlaneGlobalAccess is whenever master is accessible globally or not. Honored when enabled is true.
+	// +optional
+	ControlPlaneGlobalAccess bool `json:"controlPlaneGlobalAccess,omitempty"`
+
+	// ControlPlanePrivateEndpoint is whether the master's internal IP address is used as the cluster endpoint.
+	// Honored when enabled is true.
+	// +optional
+	ControlPlanePrivateEndpoint bool `json:"controlPlanePrivateEndpoint,omitempty"`
+
+	// DisableDefaultSNAT is disables cluster default sNAT rules. Honored when enabled is true.
+	// +optional
+	DisableDefaultSNAT bool `json:"disableDefaultSNAT,omitempty"`
+}
+
+// AuthorizedNetworkCidrBlock defines the CIDR block configuration.
+type AuthorizedNetworkCidrBlock struct {
+	// Name is the name of the CIDR Block.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// CidrBlock is the list of CIDR blocks associated with Authorized Network.
+	// +optional
+	CidrBlock map[string]string `json:"cidrBlock,omitempty"`
+}
+
+// AuthorizedNetwork provide an IP-based firewall that controls access to the GKE control plane.
+type AuthorizedNetwork struct {
+	// Enabled defines Whether enable authorized network or not.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
That's correct, @akshay196 what do you think?

Reply:
@richardcase yes, you're correct. We could remove enable and rely on whether `authorizedNetwork` is specified or not. However separate `Enabled` field would allow user to directly enable/disable authorized network without need to remove cidr blocks.
I am fine either ways.